presto-docs/src/main/sphinx/connector/memory.rst

```diff
@@ -0,0 +1,67 @@
+================
+Memory Connector
+================
+
+The Memory connector stores all data and metadata in RAM on workers
+and both are discarded when Presto restarts.
+
+.. warning::
+
+    This connector is in early experimental stage it is not recommended
+    to use it in a production environment. The intended use of this
+    connector at this point is purely for testing.
+
+.. warning::
+
+    After ``DROP TABLE`` memory is not released immediately. It is released
+    after next write access to memory connector.
```

Comment:
I should have phrased my question more clearly. I'm specifically asking about why memory is not released on next `select`. I'm not against this. I just want to understand the decision here as this appears like a conscious decision.